src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -26,11 +26,9 @@ private BoundExpression BindAwait(BoundExpression expression, CSharpSyntaxNode n
             PropertySymbol isCompleted = null;
             MethodSymbol getResult = null;
 
-            bool hasErrors = false;
-
-            hasErrors =
-                ReportBadAwaitWithoutAsync(node, diagnostics) |
-                ReportBadAwaitContext(node, diagnostics) |
+            bool hasErrors =
+                ReportBadAwaitWithoutAsync(node, diagnostics) ||
+                ReportBadAwaitContext(node, diagnostics) ||
```

Comment:
Will this change cause issues now that diagnostics may not have errors populated by ReportBadAwaitContext  if the first condition returns true?